lighthouse-core/gather/gather-runner.js

```diff
@@ -119,6 +121,53 @@ class GatherRunner {
     return {};
   }
 
+  /**
+   * Rejects if any open tabs would share a service worker with the target URL.
+   * This includes the target tab, so navigation to something like about:blank
+   * should be done before calling.
+   * @param {LH.Gatherer.FRProtocolSession} session
+   * @param {string} pageUrl
+   * @return {Promise<void>}
+   */
+  static assertNoSameOriginServiceWorkerClients(session, pageUrl) {
```

Comment:
I moved this *here* instead of service-workers because it's going to be specific to legacy flow and not shared. We can't assert this in FR where we'll be *on* the page without any `about:blank` in between.

Given our neutering of the PWA checks I'm not sure it really has much value for us today anyway.